### ISSUE TITLE
Using the same $ref multiple times in a single response causes collisions in generated code

### ISSUE THREAD

**Issue by bradj** [aarondl/oa3#4]
The following path defines 2 responses which use the same `$ref`:

```yaml
/users:
    get:
      operationId: getUsers
      summary: Get all user objects
      tags: ["auth"]
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Users"
        "400":
          description: User error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/GeneralError"
        "500":
          description: Server Error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/GeneralError"
```

The generated code creates 2 identical function definitions:

```golang
type GetUsersResponse interface {
	GetUsersImpl()
}

// GetUsersImpl implements GetUsersResponse(200) for Users
func (Users) GetUsersImpl() {}

// GetUsersImpl implements GetUsersResponse(400) for GeneralError
func (GeneralError) GetUsersImpl() {}

// GetUsersImpl implements GetUsersResponse(500) for GeneralError
func (GeneralError) GetUsersImpl() {}
```

Which generates the following build error:

`GeneralError.GetUsersResponse redeclared in this block`

**Comment by aarondl** [comment]
Implementation limitation at the moment unfortunately :(

Because the response code is implied by the response object you return, there's no way to tell the oa3 generated code that you want a specific response code.

Changing this would fundamentally alter the library in a backwards-incompatible way and make it more cumbersome to use. For now I'd suggest looking at alternative ways of structuring the API, it could be argued that there's little merit returning the same response type for multiple response codes.

**Comment by aarondl** [comment]
This is resolved by commit 7726910

See changelog for details.
